packages/cozy-sharing/src/index.jsx

```diff
@@ -213,6 +218,27 @@ export class SharingProvider extends Component {
     return resp
   }
 
+  updateSharingLinkPermissions = async (document, newVerbs) => {
+    const permissions = getDocumentPermissions(this.state, document.id)
+
+    const responses = await Promise.all(
+      permissions.map(async permissionDocument => {
+        const updatedPermissions = permissionDocument.attributes.permissions
+        Object.keys(updatedPermissions).map(permType => {
+          updatedPermissions[permType].verbs = newVerbs
+        })
+
+        const resp = await this.props.client
+          .collection('io.cozy.permissions')
+          .add(permissionDocument, updatedPermissions)
+        this.dispatch(updateSharingLink(resp))
+        return resp
+      })
+    )
+
+    return responses[0]
+  }
```

Comment:
We handle several permissions, but we only return the result of the first one?

Reply:
Also this method is more generic than just "updateSharingLinkPermissions". It's more "updateDocumentPermissions", no ?

Reply:
No catch ?

Reply:
> We handle several permissions, but we only return the result of the first one?

Since having one document with multiple permissions is not a "normal" state, I didn't want to propagate it too much down the chain. But tthe other methods do it as well, so I'll update this one to stay consistent.

> Also this method is more generic than just "updateSharingLinkPermissions". It's more "updateDocumentPermissions", no ?

Yes, you're right.

> No catch ?

Not here… but actually there should be one in `ShareByLink`, i'll add it.